app/src/lib/stores/pull-request-store.ts

```diff
@@ -13,136 +13,181 @@ import {
-      if (headId) {
-        head = await this.repositoriesStore.findGitHubRepositoryByID(headId)
+      const apiStatus = await api.fetchCombinedRefStatus(
+        repository.owner.login,
+        repository.name,
+        pr.head.sha
+      )
+
+      const status = {
+        pullRequestNumber: pr.number,
+        state: apiStatus.state,
+        totalCount: apiStatus.total_count,
+        sha: pr.head.sha,
       }
 
-      // We know the base repo ID can't be null since it's the repository we
-      // fetched the PR from in the first place.
-      const baseId = forceUnwrap(
-        'PR cannot have a null base repo id',
-        pr.base.repoId
-      )
-      const base = forceUnwrap(
-        'PR cannot have a null base repo',
-        await this.repositoriesStore.findGitHubRepositoryByID(baseId)
-      )
+      statuses.push({
+        pullRequestId: pr.id,
+        state: apiStatus.state,
+        totalCount: apiStatus.total_count,
+        sha: pr.head.sha,
+      })
 
-      // We can be certain the PR ID is valid since we just got it from the
-      // database.
-      const prID = forceUnwrap(
-        'PR cannot have a null ID after being retrieved from the database',
-        pr.id
+      prs.push(
+        new PullRequest(
+          pr.id,
+          pr.created,
+          status,
+          pr.title,
+          pr.number,
+          pr.head,
+          pr.base,
+          pr.author
+        )
       )
-      const prStatus = await this.getPullRequestStatusById(pr.head.sha, prID)
+    }
 
-      const builtPR = new PullRequest(
-        prID,
-        new Date(pr.createdAt),
-        prStatus,
-        pr.title,
-        pr.number,
-        new PullRequestRef(pr.head.ref, pr.head.sha, head),
-        new PullRequestRef(pr.base.ref, pr.base.sha, base),
-        pr.author
-      )
+    await this.writePRStatus(statuses)
+
+    this.pullRequestStatuses = await this.getPRStatuses(
+      repository,
+      pullRequests
+    )
+    this.emitUpdate()
+  }
+
+  private async getPRStatusById(
+    sha: string,
+    pullRequestId: number
+  ): Promise<PullRequestStatus | null> {
+    const result = await this.pullRequestDatabase.pullRequestStatus
+      .where('[sha+pullRequestId]')
+      .equals([sha, pullRequestId])
+      .limit(1)
+      .first()
 
-      builtPullRequests.push(builtPR)
+    if (!result) {
+      return null
     }
 
-    return builtPullRequests
+    return new PullRequestStatus(
+      result.pullRequestId,
+      result.state,
+      result.totalCount,
+      result.sha
+    )
   }
 
-  private async writePullRequests(
+  private async writePRs(
     pullRequests: ReadonlyArray<IAPIPullRequest>,
     repository: GitHubRepository
-  ): Promise<void> {
+  ): Promise<number> {
     const repoId = repository.dbID
 
     if (!repoId) {
       fatalError(
         "Cannot store pull requests for a repository that hasn't been inserted into the database!"
       )
 
-      return
+      return -1
```

Comment:
`fatalError` will throw as a side-effect above, so while this code compiles we won't ever be able to handle the `-1` case. If we care about this code path, let's just `log.error` above.

Reply:
I only put the `-1` so that it would compile since `fatalError` returns `Never`. I don't really care about the code path, but I'm not sure how to not return and still appease TS.

Reply:
:thumbsup: so given the number isn't really interesting, perhaps we could make this method return `Promise<boolean>`? This means you could just `return false` here, `return true` further down, and the caller would look a bit cleaner:

```ts
const result = await this.writePRs(raw, repository)
if (!result) {
  return
}
```

Reply:
sure thing, we can return void too, I only chose a number in case we wanted to know how many elements were inserted. Probably best to call a YAGNI it here and just return `void`